src/components/SimpleSlider.js

```diff
@@ -0,0 +1,173 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const Radium = require('radium');
+const _merge = require('lodash/merge');
+
+const StyleConstants = require('../constants/Style');
+
+const SimpleSlider = React.createClass({
+  propTypes: {
+    disabled: React.PropTypes.bool,
+    onPercentChange: React.PropTypes.func.isRequired,
+    percent: React.PropTypes.number.isRequired,
+    selectedColor: React.PropTypes.string,
+    styles: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      disabled: false,
+      selectedColor: StyleConstants.Colors.PRIMARY
+    };
+  },
+
+  getInitialState () {
+    return {
+      dragging: false,
+      leftPixels: 0,
+      width: 0
+    };
+  },
+
+  componentDidMount () {
+    this._setDefaults();
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (this.props.percent !== newProps.percent) {
+      const leftPixels = newProps.percent * this.state.width;
+
+      this.setState({ leftPixels });
+    }
+  },
+
+  _setDefaults () {
```

Comment:
What do you think about removing this function and then moving its content directly into `componentDidMount` method? I think it is easier to see what is going on.